python/cudf/cudf/core/series.py

```diff
@@ -166,6 +166,16 @@ def __init__(
             if dtype is not None:
                 data = data.astype(dtype)
 
+        if isinstance(data, dict):
+            if any(is_list_like(value) for value in data.values()):
+                raise NotImplementedError(
+                    "Nested lists are not yet supported in cudf."
+                )
```

Comment:
If someone gives us a large dictionary this is expensive. What happens if we don't do this check? It would presumably throw somewhere in `as_column`?

Reply:
Yeah, removed this as we seem to throw in `as_column`
```python
>>> import cudf
>>> s = cudf.Series({1:[1, 2, 3]})
Traceback (most recent call last):
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.7/site-packages/cudf/core/column/column.py", line 1581, in as_column
    memoryview(arbitrary), dtype=dtype, nan_as_null=nan_as_null
TypeError: memoryview: a bytes-like object is required, not 'dict_values'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.7/site-packages/cudf/core/series.py", line 176, in __init__
    data.values(), nan_as_null=nan_as_null, dtype=dtype
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.7/site-packages/cudf/core/column/column.py", line 1606, in as_column
    nan_as_null=nan_as_null,
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.7/site-packages/cudf/core/column/column.py", line 1402, in as_column
    "cudf doesn't support list like data types"
NotImplementedError: cudf doesn't support list like data types
```